scenarios/aws/microVMs/microvms/libvirt.go

```diff
@@ -177,39 +178,59 @@ func (vm *VMCollection) SetupCollectionDomainConfigurations(depends []pulumi.Res
 			}
 		}
 
-		vm.domains = append(vm.domains, domains...)
+		if _, ok := vm.domains[set.ID]; ok {
+			vm.domains[set.ID] = append(vm.domains[set.ID], domains...)
+		} else {
+			vm.domains[set.ID] = domains
+		}
```

Comment:
nit: I think this could be simplified

```suggestion
		vm.domains[set.ID] = append(vm.domains[set.ID], domains...)
```